common/src/main/java/com/wynntils/features/user/redirects/BlacksmithRedirectFeature.java

```diff
@@ -0,0 +1,112 @@
+        event.setCanceled(true);
+
+        EnumMap<ItemTier, Integer> totalItems = new EnumMap<>(ItemTier.class);
+        // How many emeralds/scrap we got from the transaction.
+        String paymentString = messageMatcher.group(3);
+        // Tracks count of sold or scrapped items
+        int totalItemInteger = 0;
+        // Full message to send to the user.
+        String sendableMessage = "";
+
+        // This is for selling items for emeralds.
+        if (messageMatcher.group(1).equals("sold me")) {
+
+            // Retrieve the color code of the item, and then match it to the item tier.
+            Matcher itemMatcher =
+                    ITEM_PATTERN.matcher(messageMatcher.group(2)); // Second group contains all of the items.
+            // Tally up the items that we sold.
+            while (itemMatcher.find()) {
+                ChatFormatting itemColor = ChatFormatting.getByCode(itemMatcher
+                        .group(1)
+                        .charAt(0)); // find the color code to find ChatFormatting and ascertain the tier.
+                ItemTier tierToIncrease = ItemTier.fromChatFormatting(itemColor);
+                if (tierToIncrease == null) continue;
+                totalItems.put(tierToIncrease, totalItems.getOrDefault(tierToIncrease, 0) + 1);
+                totalItemInteger++;
+            }
+            String itemPluralizer = totalItemInteger == 1 ? "item" : "items";
+
+            // Build up the string that outlines how many items were sold in what tier (0/0/0/0/0/0/0/0).
+            StringBuilder countByTier = new StringBuilder();
+            for (ItemTier tier : ItemTier.values()) {
+                countByTier.append('/' + tier.getChatFormatting().toString() + totalItems.getOrDefault(tier, 0));
+                countByTier.append(ChatFormatting.LIGHT_PURPLE);
+            }
+            countByTier.append(")");
+            countByTier.setCharAt(0, '(');
+
+            String countByTierString = countByTier.toString();
+
+            // Sold 1 (1/0/0/0/0/0/0/0) item for 4e.
+            sendableMessage = String.format(
+                    "%sSold %d %s %s for %s%s%s%s.",
+                    ChatFormatting.LIGHT_PURPLE,
+                    totalItemInteger,
+                    itemPluralizer,
+                    countByTierString,
+                    ChatFormatting.GREEN,
+                    paymentString,
+                    EmeraldSymbols.EMERALDS,
+                    ChatFormatting.LIGHT_PURPLE);
+        }
+        // Scrapping items for scrap.
+        else {
+            Matcher itemMatcher =
+                    ITEM_PATTERN.matcher(messageMatcher.group(2)); // Second group contains all of the items.
+            while (itemMatcher.find()) {
+                totalItemInteger++;
+            }
+            String itemPluralizer = totalItemInteger == 1 ? "" : "s";
+
+            sendableMessage = String.format(
+                    "%sScrapped %d %s for %s%s scrap%s.",
```

Comment:
Actually, is there a reason why we  don't have item tier breakdown here? Even if legacy didnt do it, I think it is nice to have it

Reply:
Feels more cluttered to me IMO, but I can look into adding it. The only technical concern I can think of it the message overwriting it's bounds, like:

"Scrapped 5 items for 16 scrap."
versus
"Scrapped 5 (0/5/0/0/0/0/0/0) items for
16 scrap"

might not matter though and we could be fine, the only concern is the relative fingerprint the sendablemessage will have

Reply:
could ignore 0s (so instead of 0/5/0/0/0/0/0/1 it would be 5/1 but color coded)